src/peer_manager.rs

```diff
@@ -342,7 +342,7 @@ impl PeerManager {
     }
 
     /// Return true if already has a candidate
-    pub fn has_candidate(&self) -> bool {
+    pub fn has_resource_proof_candidate(&self) -> bool {
```

Comment:
I'm not sure about this rename. I think "candidate" here means someone who is aspiring to join a section and "resource proof" is what they have to perform to do that. `resource_proof_candidate` sounds like "resource proof" itself is the thing they are aspiring to.

Reply:
Well they are aspiring to "resource proof", initially. If they pass resource proof, the section will take them in. I rather keep a less generic name than has_candidate, but I'm open to other more specific names.

Reply:
The way I see it is that we can have multiple candidates aspiring to become section members but only one of those is selected for the resource proof at a time. So all except the one currently undergoing the proof can be considered "resource proof candidates", because they are waiting for the resource proof. The other one is not a candidate for resource proof anymore because they are already undergoing it. So maybe this should be `member_candidate`?

Anyway, I thiks this is a very minor semantic issue and probably not worth spending that much time on, so I'll leave this up to you.

Reply:
I kind of see where you are going, but I'm not sure `member_candidate` is really clear.
Because the Candidate enum is all about resource proof, I'll keep that attached to the name for now.
This code will change a lot in the close future as things need to eventually move in the chain and follow the designed flow, so it is something we can and should address when things have cleaned up a bit.

Reply:
OK, fair enough. Let's keep it.